k9mail/src/main/java/com/fsck/k9/activity/RecipientAdapter.java

```diff
@@ -0,0 +1,166 @@
+            view = newView(parent);
+        }
+        Recipient recipient = getItem(position);
+        bindView(view, recipient);
+
+        return view;
+    }
+
+    public View newView(ViewGroup parent) {
+        View view = LayoutInflater.from(context).inflate(R.layout.recipient_dropdown_item, parent, false);
+        RecipientTokenHolder holder = new RecipientTokenHolder(view);
+        view.setTag(holder);
+        return view;
+    }
+
+    public void bindView(View view, Recipient recipient) {
+
+        RecipientTokenHolder holder = (RecipientTokenHolder) view.getTag();
+
+        holder.name.setText(recipient.address.getPersonal());
+
+        String address = recipient.address.getAddress();
+        holder.email.setText(address);
+
+        setContactPhotoOrPlaceholder(context, holder.photo, recipient);
+
+        if (!showCryptoStatus) {
+            holder.cryptoStatus.setVisibility(View.GONE);
+            return;
+        }
+        holder.cryptoStatus.setVisibility(View.VISIBLE);
+
+        Integer cryptoStatusRes = null, cryptoStatusColor = null;
+        RecipientCryptoStatus cryptoStatus = recipient.getCryptoStatus();
+        switch (cryptoStatus) {
+            case AVAILABLE_TRUSTED:
+                cryptoStatusRes = R.drawable.status_lock_closed;
+                cryptoStatusColor = context.getResources().getColor(R.color.openpgp_green);
+                break;
+            case AVAILABLE_UNTRUSTED:
+                cryptoStatusRes = R.drawable.status_lock_error;
+                cryptoStatusColor = context.getResources().getColor(R.color.openpgp_orange);
+                break;
+            case UNAVAILABLE:
+                cryptoStatusRes = R.drawable.status_lock_open;
+                cryptoStatusColor = context.getResources().getColor(R.color.openpgp_red);
+                break;
+        }
+
+        if (cryptoStatusRes != null) {
+            // we could do this easier with setImageTintList, but that's API level 21
+            Drawable drawable = context.getResources().getDrawable(cryptoStatusRes);
+            // noinspection ConstantConditions, we know the resource exists!
+            drawable.mutate();
+            drawable.setColorFilter(cryptoStatusColor, Mode.SRC_ATOP);
+            holder.cryptoStatus.setImageDrawable(drawable);
+            holder.cryptoStatus.setVisibility(View.VISIBLE);
+        } else {
+            holder.cryptoStatus.setVisibility(View.GONE);
+        }
+
+    }
+
+    public static void setContactPhotoOrPlaceholder(Context context, ImageView imageView, Recipient recipient) {
+        imageView.setImageDrawable(null);
+        if (recipient.photoThumbnailUri != null) {
+            Glide.with(context).load(recipient.photoThumbnailUri).into(imageView);
+        } else {
+            ContactPicture.getContactPictureLoader(context).loadContactPicture(recipient.address, imageView);
+        }
+    }
+
+    @Override
+    public Filter getFilter() {
+        return new Filter() {
```

Comment:
yes, but a working `Filterable` implementation is required by `AutoCompleteTextView`, even if we don't do anything with it because we reload the data in a loader in `performFiltering` everytime